### PR TITLE
Upgrade packages that use old versions of graceful-fs

### PR DESCRIPTION
**Pull request by thomasjo**
Bumps packages that have dependencies on graceful-fs to versions that uses graceful-fs v4 or higher. 

This is **necessary** for Node.js v6 support.

TODO:
- [x] Bump fstream.
- [x] Open PRs against [less-cache](https://github.com/atom/less-cache) — https://github.com/atom/less-cache/pull/10
  - [x] Bump less-cache once PRs have shipped.
- [ ] Fix regression during the `prebuild-less` task. See https://github.com/atom/atom/pull/12176#issuecomment-240339043

### CODE REVIEW

**Comment by thomasjo** [comment]
@atom/feedback I think this should be good to go. Would appreciate some 👀 and maybe a bit of 🚬 testing.

**Comment by thomasjo** [comment]
Nevermind, turns out this causes a problem during `script/build` (and possibly elsewhere)

```
Running "prebuild-less:src" (prebuild-less) task
Warning: variable @fold is undefined Use --force to continue.

Aborted due to warnings.
```

**Comment by maxbrunsfeld** [comment]
> this causes a problem during `script/build`

I've tracked this problem down to what I believe is a bug in `less`. I have opened a PR fixing the bug: https://github.com/less/less.js/pull/2953

File: static/variables/octicon-mixins.less [hunk 1]
```diff
@@ -1,3 +1,5 @@
+@import "octicon-utf-codes.less";
+
```
Comment (maxbrunsfeld) [inline static/variables/octicon-mixins.less]: See https://github.com/less/less.js/pull/2953

**Comment by maxbrunsfeld** [comment]
Interesting; the janky build passed, but the circle build failed with

```
Expected '#header {
  color: #4D926F;
}
h2 {
  color: #4D926F;
}
' to be '#header {
  color: #4d926f;
}
h2 {
  color: #4d926f;
}
'
```

Maybe the new version of `less` normalizes color literals to always use capital letters?

**Comment by maxbrunsfeld** [comment]
OK, it turns out that the less bug I mentioned above is the expected behavior, so I've changed our base stylesheets to avoid imports in nested scopes. Now I've run into another less bug:

https://github.com/less/less.js/pull/2955

**Pull request closed without merging**